File: src/main/java/com/github/t1/yaml/model/Collection.java
package com.github.t1.yaml.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import static com.github.t1.yaml.model.Collection.Style.BLOCK;

@EqualsAndHashCode(callSuper = true)
public abstract @Data class Collection extends Node {
    public enum Style {
        FLOW, BLOCK
    }

    private Style style = BLOCK;
}

File: src/main/java/com/github/t1/yaml/model/Collection.kt
package com.github.t1.yaml.model

import com.github.t1.yaml.model.Collection.Style.BLOCK

interface Collection : Node {
    var style: Style

    enum class Style {
        FLOW, BLOCK
    }

    companion object {
        val DEFAULT_STYLE = BLOCK
    }
}
